cpp/src/arrow/compute/kernels/codegen_internal.h

```diff
@@ -1169,6 +1169,19 @@ ArrayKernelExec GeneratePhysicalNumeric(detail::GetTypeId get_id) {
   }
 }
 
+template <template <typename... Args> class Generator, typename... Args>
+ArrayKernelExec GeneratePhysicalDecimal(detail::GetTypeId get_id) {
+  switch (get_id.id) {
```

Comment:
There are two kinds of kernel generator dispatchers, each with an "almost" name convention:
* Fixed output type, set via `typename Type0` --> named `GenerateXXX`
* Output type same as input type (do not have `typename Type0`) --> named `GenerateXXXToXXX`

Based on this, I suggest to rename this function to `GeneratePhysicalDecimalToPhysicalDecimal`.

Reply:
Done.